workflow/controller/operator_test.go

```diff
@@ -1957,6 +1957,59 @@ func TestSuspendWithDeadline(t *testing.T) {
 	assert.True(t, found)
 }
 
+var suspendTemplateInputResolution = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: suspend-template
+spec:
+  entrypoint: suspend
+  templates:
+  - name: suspend
+    inputs:
+        parameters:
+            - name: param1
+              value: "{\"enum\": [\"one\", \"two\", \"three\"]}"
+            - name: param2
+              value: "value2"
+    suspend: {}
+`
+
+func TestSuspendInputsResolution(t *testing.T) {
+	cancel, controller := newController()
+	defer cancel()
+	wfcset := controller.wfclientset.ArgoprojV1alpha1().Workflows("")
+
+	ctx := context.Background()
+	wf := wfv1.MustUnmarshalWorkflow(suspendTemplateInputResolution)
+	wf, err := wfcset.Create(ctx, wf, metav1.CreateOptions{})
```

Comment:
you don't need to have `wfcset` in this test at all,  you can access `woc.wf`

Reply:
awesome, less code